### PR TITLE
Taskmaster

### PR DESCRIPTION
**Pull request by gonced8**
Extended to support MT5.
Upgraded some packages to support this transformer and whenever necessary for compatibility.

### CODE REVIEW

File: cli.py [hunk 1]
```diff
@@ -129,15 +140,20 @@ def test(
     answers and produce replies.
     """
     logging.disable(logging.WARNING)
-    hparams_file = experiment + "hparams.yaml"
+    hparams_file = os.path.join(experiment, "hparams.yaml")
```
Comment (gonced8) [inline cli.py:143]: So that a user can use as argument "path_to_folder" instead of "path_to_folder/"

File: configs/taskmaster-gpt2.yaml [hunk 2]
```diff
@@ -30,7 +30,7 @@ max_epochs: 2                             # Max number of epochs
 # --------------------------------- Dataset -------------------------------------------------------
 pretrained_model: gpt2
 train_data: data/polyglot-taskmaster/TM-en-train.json
-valid_data: data/polyglot-taskmaster/TM-en-train.json
+valid_data: data/polyglot-taskmaster/TM-en-dev.json
```
Comment (gonced8) [inline configs/taskmaster-gpt2.yaml:33]: GPT2 was using the same training set for validation

File: configs/taskmaster-t5.yaml [hunk 3]
```diff
@@ -36,9 +36,9 @@ num_candidates: 5                         # Number of distractors
 
 # -------------------------------- T5 Fine-tuning -----------------------------------------------
 model: AssistantT5
-pretrained_model: google/t5-small
+pretrained_model: t5-small
```
Comment (gonced8) [inline configs/taskmaster-t5.yaml:39]: Hugging Face changed google/t5-small to t5-small

File: models/gpt2/model.py [hunk 4]
```diff
@@ -11,7 +11,7 @@
 import torch
 import yaml
 from transformers import AdamW, GPT2DoubleHeadsModel
-from transformers.modeling_gpt2 import GPT2DoubleHeadsModelOutput
+from transformers.models.gpt2.modeling_gpt2 import GPT2DoubleHeadsModelOutput
```
Comment (gonced8) [inline models/gpt2/model.py:14]: Had to change due to upgrading the transformers package

File: requirements.txt [hunk 5]
```diff
@@ -1,6 +1,7 @@
...
+transformers==4.3.3
 pytorch-lightning==0.9.0
 pytorch-nlp==0.5.0
 sacrebleu==1.4.14
-bert-score==0.3.6
-nltk==3.5
\ No newline at end of file
+bert-score==0.3.7
+nltk==3.5
+sentencepiece==0.1.95
```
Comment (gonced8) [inline requirements.txt:7]: Has to be included for transformers>v.4

**Review by gonced8 — COMMENTED** [review]
Changed some cases to use os.path.join instead of concatenating strings.
GPT2 validation set from TM-en-train to TM-en-dev.
t5-small model name changed in Hugging Face.
Upgraded some packages to support the MT5 model and some others for compatibility.
Extended to support MT5.
From the tests I performed, GPT2 and T5 seem to still work as expected.
